Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -2697,15 +2712,33 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     if (!webKitRules)
         return;
 
-    [declarativeNetRequestRuleStore() compileContentRuleListForIdentifier:uniqueIdentifier() encodedContentRuleList:webKitRules completionHandler:^(WKContentRuleList *ruleList, NSError *error) {
-        if (error) {
-            RELEASE_LOG_ERROR(Extensions, "Error compiling declarativeNetRequest rules: %{public}@", privacyPreservingDescription(error));
-            return;
+    auto *previouslyLoadedHash = objectForKey<NSString>(m_state, lastLoadedDNRHashStateKey);
+    auto *hashOfWebKitRules = computeStringHashForContentBlockerRules(webKitRules);
+
+    [declarativeNetRequestRuleStore() lookUpContentRuleListForIdentifier:uniqueIdentifier() completionHandler:^(WKContentRuleList *foundRuleList, NSError *) {
+        if (foundRuleList) {
```

Comment:
should we be logging an error if the lookup failed?

Reply:
That can happen for normal reasons, like if it's the first time an extension was loaded, or if the user deleted the file on disk.